### PR TITLE
refactor and move to 32-bit protected mode

### PR DESCRIPTION
**Pull request by botondmester**

### CODE REVIEW

File: kernel/memory/gdt.c [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+              0xFFFFF,
+              GDT_ACCESS_PRESENT | GDT_ACCESS_RING0 | GDT_ACCESS_CODE_SEGMENT | GDT_ACCESS_CODE_READABLE,
+              GDT_FLAG_64BIT | GDT_FLAG_GRANULARITY_4K),
+    
+    // Kernel 64-bit data segment
+    GDT_ENTRY(0,
+              0xFFFFF,
+              GDT_ACCESS_PRESENT | GDT_ACCESS_RING0 | GDT_ACCESS_DATA_SEGMENT | GDT_ACCESS_DATA_WRITEABLE,
+              GDT_FLAG_64BIT | GDT_FLAG_GRANULARITY_4K),
+};
```
Comment (KnedlikMCPE) [inline kernel/memory/gdt.c:80]: Missing User Mode segments and Task State Segment

File: kernel/print.c [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            return;
+        }
+
+        print_char(character);
+    }
+}
+
+void print_set_color(uint8_t foreground, uint8_t background) {
+    color = foreground + (background << 4);
+}
```
Comment (KnedlikMCPE) [inline kernel/print.c:83]: I'm once again asking for _Video memory location_

**Review by KnedlikMCPE — CHANGES_REQUESTED** [review]

File: kernel/memory/gdt.c (hunk 1 above)
Comment (botondmester) [reply kernel/memory/gdt.c:80]: We will add that later

File: kernel/memory/gdta.asm [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+    ; reload code segment
+    mov eax, [ebp + 12]
+    push eax
+    push .reload_cs
+    retf
+
+    .reload_cs:
+        ; reload data segments
+        mov ax, [ebp + 16]
```
Comment (KnedlikMCPE) [inline kernel/memory/gdta.asm:23]: 16-bit registers

File: kernel/print.c (hunk 2 above)
Comment (botondmester) [reply kernel/print.c:83]: Comment added, but not commited yet, check discord

**Pull request merged**